app/lib/efile/md/two_income_subtraction_worksheet.rb

```diff
@@ -0,0 +1,132 @@
+        set_line(:MD_TWO_INCOME_WK_LINE_3_B, -> { calculate_line_3 :spouse })
+        set_line(:MD_TWO_INCOME_WK_LINE_4_B, -> { calculate_line_4 :spouse })
+        set_line(:MD_TWO_INCOME_WK_LINE_5_B, -> { calculate_line_5 :spouse })
+        set_line(:MD_TWO_INCOME_WK_LINE_6, :calculate_line_6)
+        set_line(:MD_TWO_INCOME_WK_LINE_7, :calculate_line_7)
+      end
+
+      def calculate_fed_income(primary_or_spouse)
+        filer_ssn = @intake.send(primary_or_spouse).ssn
+
+        wage_income = @direct_file_data.w2s
+          .select { |w2| w2.EmployeeSSN == filer_ssn }
+          .sum { |w2| w2.WagesAmt&.round }
+
+        interest_income = @direct_file_json_data.interest_reports
+          .select { |interest_report| interest_report.recipient_tin.delete("-") == filer_ssn }
+          .sum { |interest_report|
+            (interest_report.amount_1099&.round || 0) + (interest_report.amount_no_1099&.round || 0)
+          }
+
+        retirement_income = @intake.state_file1099_rs
+          .select { |form1099r| form1099r.recipient_ssn == filer_ssn }
+          .sum { |form1099r| form1099r.taxable_amount&.round }
+
+        unemployment_income = @intake.state_file1099_gs
+          .select { |form1099g| form1099g.recipient.to_sym == primary_or_spouse }
+          .sum { |form1099g| form1099g.unemployment_compensation_amount&.round }
+
+        wage_income +
+          interest_income +
+          retirement_income +
+          unemployment_income
+      end
+
+      def calculate_fed_subtractions(primary_or_spouse)
+        filer_json = @direct_file_json_data.filers
+          .find { |df_filer_data|
+            df_filer_data.tin.delete("-") == @intake.send(primary_or_spouse).ssn
+          }
+
+        # TODO: Some MFJ tests are missing spouse JSON - should not happen in prod
+        return 0 unless filer_json
+
+        student_loan_interest = {
+          primary: @intake.primary_student_loan_interest_ded_amount&.round,
+          spouse: @intake.spouse_student_loan_interest_ded_amount&.round,
+        }[primary_or_spouse]
+
+        student_loan_interest +
+          filer_json.educator_expenses&.round +
+          filer_json.hsa_total_deductible_amount&.round
+      end
+
+      private
+
+      def calculate_line_1(primary_or_spouse)
+        calculate_fed_income(primary_or_spouse) - calculate_fed_subtractions(primary_or_spouse)
+      end
+
+      def calculate_line_2(primary_or_spouse)
+        @intake.state_file_w2s
+               .select { |w2| w2.employee_ssn == @intake.send(primary_or_spouse).ssn }
+               .sum { |w2| w2.box14_stpickup&.round || 0 }
+      end
+
+      def calculate_line_3(primary_or_spouse)
+        if primary_or_spouse == :primary
+          @lines[:MD_TWO_INCOME_WK_LINE_1_A].value +
+            @lines[:MD_TWO_INCOME_WK_LINE_2_A].value
+        else
+          @lines[:MD_TWO_INCOME_WK_LINE_1_B].value +
+            @lines[:MD_TWO_INCOME_WK_LINE_2_B].value
+        end
```

Comment:
```suggestion
        filer = primary_or_spouse == :primary ? "A" : "B"
         line_or_zero("MD_TWO_INCOME_WK_LINE_1_#{filer}") +
            line_or_zero("MD_TWO_INCOME_WK_LINE_2_{filer}")
```
we spoke during review about changing this method to look something like this to make it more clear that the calculations are very similar. 

i think it is more clear this way, but i will also leave it up to you since it feels more stylistic.